docs/overview/whatisvpp/softwarearchitecture.rst

```diff
@@ -0,0 +1,707 @@
+.. _softwarearchitecture:
+
+.. toctree::
+
+Software Architecture
+=====================
+
+The fd.io vpp implementation is a third-generation vector packet
+processing implementation specifically related to US Patent 7,961,636,
+as well as earlier work. Note that the Apache-2 license specifically
+grants non-exclusive patent licenses; we mention this patent as a
+point of historical interest.
+
+For performance, the vpp dataplane consists of a directed graph of
+forwarding nodes which process multiple packets per invocation. This
+schema enables a variety of cache-related optimizations: pipelining
+and prefetching to cover dependent read latency, inherent I-cache
+phase behavior. Aside from hardware input and hardware output nodes,
+the entire forwarding graph is portable code.
+
+Depending on the scenario at hand, we often spin up multiple worker
+threads which process ingress-hashes packets from multiple queues using
+identical forwarding graph replicas. 
+
+Implemetation taxonomy
+----------------------
+
+The vpp dataplane consists of four distinct layers: 
+
+* An infrastructure layer comprising vppinfra, vlib, svm, and binary api libraries. See .../src/{vppinfra, vlib, svm, vlibapi, vlibmemory}
+
+* A generic network stack layer: vnet. See .../src/vnet
```

Comment:
Would be more meaningful as a links to doxygen. 
I don't see a nice way to link to a specific directory - a link to the top level will suffice for the moment.

`vnet <https://docs.fd.io/vpp/18.04/>`_